src/dbnode/storage/limits/query_limits.go

```diff
@@ -106,6 +108,10 @@ func NewQueryLimits(options Options) (QueryLimits, error) {
 			sourceLoggerBuilder, map[string]string{"type": "aggregate"})
 	)
 
+	// NB: All metrics for aggregate methods should show up with the same metric
```

Comment:
why don't we just stop updating `docsMatched` for the aggregate path so it essentially becomes docs matched for the fetch path.

also, don't follow why you need to update the name after constructing.

Reply:
We want them to appear in the same metric, but under different tags, so the generated metrics for both aggregate and fetch limits create metrics here. The name is updated to affect the error message received when the limit is tripped